### PR TITLE
Rspec2

### PR DESCRIPTION
**Pull request by franciscoabalan**
## Summary
Add test to index and show endpoints, from book resource. 

## Trellocard
https://trello.com/c/H176XSJo/22-crear-tests-para-rutas-index-y-show-de-book

## Screenshots
Captura de Pantalla 2019-10-10 a la(s) 13 35 08

Captura de Pantalla 2019-10-10 a la(s) 15 57 03

Captura de Pantalla 2019-10-10 a la(s) 17 26 53

Captura de Pantalla 2019-10-14 a la(s) 10 37 59

### CODE REVIEW

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (joredudiaz) [inline spec/controllers/api/v1/books_controller_spec.rb]: also test, when you pass a book id that does not exist or is invalid

Comment (joredudiaz) [inline spec/controllers/api/v1/books_controller_spec.rb]: also test when you param is missing

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (joredudiaz) [inline spec/controllers/api/v1/books_controller_spec.rb]: let is `lazy loading` this means that, at the moment you call the variable, the books objects array is created, so we don't need to add the bang (!). Please remove it.

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (joredudiaz) [inline spec/controllers/api/v1/books_controller_spec.rb]: also here (!) ☝️

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (franciscoabalan) [reply spec/controllers/api/v1/books_controller_spec.rb]: Done

Comment (franciscoabalan) [reply spec/controllers/api/v1/books_controller_spec.rb]: Done

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (franciscoabalan) [reply spec/controllers/api/v1/books_controller_spec.rb]: Done

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (franciscoabalan) [reply spec/controllers/api/v1/books_controller_spec.rb]: Done

**Review by joredudiaz — APPROVED** [review]
Great job! 😄 , let's wait for another approval.

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (aleochoam) [inline spec/controllers/api/v1/books_controller_spec.rb]: Rspec know this is a controller test because it is in the controller folder, so you can delete the  explicit type

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (aleochoam) [inline spec/controllers/api/v1/books_controller_spec.rb]: If you think about it, this is your subject, so instead  of putting it in a before you can make a subjet like this:

```
subject(:http_request) { get :index }
```

And change the response in the examples

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (aleochoam) [inline spec/controllers/api/v1/books_controller_spec.rb]: The subject should be at `describe` level, and you change the way it run by changing the params in the  context. In other words, this is your subject:

```
subject(:http_request) { get :show, params { id: id } }
```

And have a  `let(:id) { some_value }` in every context

**Review by aleochoam — CHANGES_REQUESTED** [review]

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (franciscoabalan) [reply spec/controllers/api/v1/books_controller_spec.rb]: Done

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (franciscoabalan) [reply spec/controllers/api/v1/books_controller_spec.rb]: DONE

File: spec/controllers/api/v1/books_controller_spec.rb
Comment (franciscoabalan) [reply spec/controllers/api/v1/books_controller_spec.rb]: DONE

**Review by aleochoam — APPROVED** [review]
Perfect! 👏

**Pull request merged**
